src/gc/gcinterface.h

```diff
@@ -174,6 +183,10 @@ enum end_no_gc_region_status
 
 typedef BOOL (* walk_fn)(Object*, void*);
 typedef void (* gen_walk_fn)(void* context, int generation, uint8_t* range_start, uint8_t* range_end, uint8_t* range_reserved);
+typedef void (* record_surv_fn)(uint8_t* begin, uint8_t* end, ptrdiff_t reloc, size_t context, BOOL compacting_p, BOOL bgc_p);
+typedef void (* fq_walk_fn)(BOOL, void*);
+typedef void (* fq_scan_fn)(Object** ppObject, ScanContext *pSC, uint32_t dwFlags);
+typedef void (* handle_scan_fn)(Object** pRef, Object* pSec, uint32_t flags, ScanContext* context, BOOL isDependent);
```

Comment:
Are the contexts here derived from the void pointers passed to the interface below? If so, I'm wondering should we be enforcing that the user passes a ScanContext by making the context argument be one instead of a void pointer, or by making these function pointers accept `void*` as a context?